x-pack/plugins/enterprise_search/public/applications/workplace_search/views/groups/groups_logic.test.ts

```diff
@@ -256,40 +250,45 @@ describe('GroupsLogic', () => {
         headers,
       };
 
+      beforeAll(() => {
+        jest.useFakeTimers();
+      });
+
+      afterAll(() => {
+        jest.useRealTimers();
+      });
+
       it('calls API and sets values', async () => {
         const setSearchResultsSpy = jest.spyOn(GroupsLogic.actions, 'setSearchResults');
-        const promise = Promise.resolve(groups);
-        http.post.mockReturnValue(promise);
+        http.post.mockReturnValue(Promise.resolve(groups));
 
         GroupsLogic.actions.getSearchResults();
-        await delay();
+        jest.advanceTimersByTime(TIMEOUT);
+        await nextTick();
         expect(http.post).toHaveBeenCalledWith('/api/workplace_search/groups/search', payload);
-        await promise;
         expect(setSearchResultsSpy).toHaveBeenCalledWith(groups);
       });
 
       it('calls API and resets pagination', async () => {
         // Set active page to 2 to confirm resetting sends the `payload` value of 1 for the current page.
         GroupsLogic.actions.setActivePage(2);
         const setSearchResultsSpy = jest.spyOn(GroupsLogic.actions, 'setSearchResults');
-        const promise = Promise.resolve(groups);
-        http.post.mockReturnValue(promise);
+        http.post.mockReturnValue(Promise.resolve(groups));
 
         GroupsLogic.actions.getSearchResults(true);
         // Account for `breakpoint` that debounces filter value.
-        await delay();
+        jest.advanceTimersByTime(TIMEOUT);
```

Comment:
Changed delay to fake timers here.

Also, fyi, there is a `delay` function available in the `@kbn/test/jest` package.